base/weakkeydict.jl

```diff
@@ -129,10 +129,12 @@ function get!(default::Callable, wkh::WeakKeyDict{K}, key) where {K}
 end
 
 function getkey(wkh::WeakKeyDict{K}, kk, default) where K
-    k = lock(wkh) do
-        k = getkey(wkh.ht, kk, nothing)
-        k === nothing && return nothing
-        return k.value
+    k = let wkh=wkh
```

Comment:
I think the problematic variable here is `k`, not `wkh`, since it's assigned both inside and outside the closure. It looks like the use inside could just be renamed.